python/lsst/ap/pipe/matchSourceInjected.py

```diff
@@ -129,8 +134,79 @@ def run(self, injectedInitialPVICat, diffIm, associatedDiaSources):
+            will be modified in place**.
+        diffIm : `lsst.afw.image.Exposure`
+            Difference image where the sources were detected.
+
+        Returns
+        -------
+        None
+        """
+        # Create a schema for the forced measurement task
+        schema = afwTable.SourceTable.makeMinimalSchema()
+        schema.addField("x", "D", "x position in image.", units="pixel")
+        schema.addField("y", "D", "y position in image.", units="pixel")
+        schema.addField("deblend_nChild", "I", "Need for minimal forced phot schema")
+
+        # Create the forced measurement task
+        forcedMeas = ForcedMeasurementTask(schema)
+        # Remove plugins that are not required for this task
+        forcedMeas.plugins.pop('base_SdssShape')
+        forcedMeas.plugins.pop('base_GaussianFlux')
+        forcedMeas.plugins.pop('base_TransformedCentroid')
+        forcedMeas.plugins.pop('base_TransformedShape')
+        # Specify the columns to copy from the input catalog to the output catalog
+        forcedMeas.copyColumns = {"coord_ra": "ra", "coord_dec": "dec"}
+
+        # Create an afw table from the input catalog
+        outputCatalog = afwTable.SourceCatalog(schema)
+        outputCatalog.reserve(len(injectedCat))
+        for row in injectedCat:
+            outputRecord = outputCatalog.addNew()
+            outputRecord.setId(row['injection_id'])
+            outputRecord.setCoord(SpherePoint(row["ra"], row["dec"], degrees))
+            outputRecord.set("x", row["x"])
+            outputRecord.set("y", row["y"])
+
+        # Generate the forced measurement catalog
+        forcedSources = forcedMeas.generateMeasCat(diffIm, outputCatalog, diffIm.getWcs())
+        # Attach the PSF shape footprints to the forced measurement catalog
+        forcedMeas.attachPsfShapeFootprints(forcedSources, diffIm)
+
+        # Copy the x and y positions from the forced measurement catalog back
+        # to the input catalog
+        for tgt, src in zip(outputCatalog, forcedSources):
+            src.set('base_SdssCentroid_x', tgt['x'])
+            src.set('base_SdssCentroid_y', tgt['y'])
+
+        # Define the centroid for the forced measurement catalog
+        forcedSources.defineCentroid('base_SdssCentroid')
+        # Run the forced measurement task
+        forcedMeas.run(forcedSources, diffIm, outputCatalog, diffIm.getWcs())
+        # Convert the forced measurement catalog to an astropy table
+        forcedSources_table = forcedSources.asAstropy()
+
+        # Add the forced measurement columns to the input catalog
+        for column in forcedSources_table.columns:
+            if "Flux" in column:
+                injectedCat["forced_"+column] = forcedSources_table[column]
+
+        # Add the SNR columns to the input catalog
+        for column in injectedCat.colnames:
+            if column.endswith("instFlux"):
+                injectedCat[column+"_SNR"] = injectedCat[column] / injectedCat[column+"Err"]
```

Comment:
Do you need to catch 0 or non-finite values in the `"Err"` columns?

Reply:
We have a not-a-number and 0 padding strategy now.